### PR TITLE
[SPARK-43876][SQL][FOLLOWUP] Add a unit test for fast hashmap for distinct queries

### PR DESCRIPTION
**Pull request by wankunde**
### What changes were proposed in this pull request?

Add UT for https://github.com/apache/spark/pull/41374 change.

### Why are the changes needed?

Add a proper UT to coverage the change.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Add UT

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 1]
```diff
@@ -1987,6 +1987,16 @@ class DataFrameAggregateSuite extends QueryTest
...
   }
+
+  test("SPARK-43876: Enable fast hashmap for distinct queries") {
+    val df = testData2.distinct()
+    checkAnswer(df, testData2)
+    val output = new java.io.ByteArrayOutputStream()
+    Console.withOut(output) {
+      df.explain("codegen")
+    }
+    assert(output.toString().contains("public class hashAgg_FastHashMap_0"))
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1998]: Nice!

**Review by dongjoon-hyun — APPROVED** [review]
cc @viirya

**Review by viirya — COMMENTED** [review]
This test looks good. I've verified that it tests the code path.

Another thing is, the claim of the proposed change is to improve distinct queries performance. But I don't see any reported number of performance. If you have run benchmark or you have production workloads getting improvement from it, could you post the numbers?

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 2]
```diff
@@ -1989,13 +1989,15 @@ class DataFrameAggregateSuite extends QueryTest
   }
 
   test("SPARK-43876: Enable fast hashmap for distinct queries") {
-    val df = testData2.distinct()
-    checkAnswer(df, testData2)
-    val output = new java.io.ByteArrayOutputStream()
-    Console.withOut(output) {
-      df.explain("codegen")
+    withSQLConf(SQLConf.ENABLE_TWOLEVEL_AGG_MAP.key -> "true") {
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1992]: I think this is enabled by default?

Comment (wankunde) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1992]: This UT will fail in SingleLevelAggregateHashMapSuite, which will disable ENABLE_TWOLEVEL_AGG_MAP. 

https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/AggregateHashMapSuite.scala#L25-L38

**Comment by wankunde** [comment]
> For the 3rd case, it becomes slower?

By default, `FAST_HASH_AGGREGATE_MAX_ROWS_CAPACITY_BIT = 16`, the fast hash map size is `1 << 16 = 65536`, and the result will be 2.0X faster. 

For the 3rd case, the fast hash map size is `1 << 20 = 1048576`, I don't know why it's slowed down.  Maybe large hash map has more cache misses.

This change will make the small queries faster.

**Comment by dongjoon-hyun** [comment]
Hi, @viirya . WDTY? Do you think you can do the final sign-off or is there other feedbacks?

**Review by viirya — APPROVED** [review]
Hmm, okay. I think previously because the condition is hard-coded so you cannot enable it through the config. Now although seems it could be slower on some cases, you can disable it.

**Comment by dongjoon-hyun** [comment]
Thank you, @wankunde and @viirya . Merged to master.

**Pull request closed without merging**